adapter/validator/src/test/java/org/mobilitydata/gtfsvalidator/validator/GtfsTypeValidatorTest.java

```diff
@@ -167,870 +218,927 @@ void outOfRangeInt_decGenerateNotice() throws IOException {
-                notice.getDescription());
+        verify(mockTimeValidator, times(1)).isValid(ArgumentMatchers.eq("01:02:03"));
     }
 
     @Test
-    void urlInCorrectFormatDoNotGenerateNotice() throws IOException {
+    void timeInIncorrectFormatGenerateNotice() {
 
-        final GtfsSpecificationProto.CsvSpecProtos testGtfsSpec = TextFormat.parse(
-                Resources.toString(Resources.getResource("validator_url_format_test_gtfs_spec.asciipb"),
-                        StandardCharsets.UTF_8),
-                GtfsSpecificationProto.CsvSpecProtos.class
-        );
+        RegexValidator mockTimeValidator = mock(RegexValidator.class);
+
+        GtfsSpecificationProto.CsvSpecProto mockFileSpec = mock(GtfsSpecificationProto.CsvSpecProto.class);
+        GtfsSpecificationProto.ColumnSpecProto mockColumnSpec = mock(GtfsSpecificationProto.ColumnSpecProto.class);
+        when(mockColumnSpec.getName()).thenReturn("time_with_regex");
+
+        GtfsSpecificationProto.ColumnInputType mockInputType = mock(GtfsSpecificationProto.ColumnInputType.class);
+        when(mockInputType.getType()).thenReturn(GtfsSpecificationProto.ColumnInputType.InputType.TIME);
 
-        GtfsTypeValidator underTest = new GtfsTypeValidator(testGtfsSpec.getCsvspec(0)
+        when(mockColumnSpec.getType()).thenReturn(mockInputType);
+        when(mockFileSpec.getColumnList()).thenReturn(List.of(mockColumnSpec));
+
+        GtfsTypeValidator underTest = new GtfsTypeValidator(mockFileSpec,
+                mock(FloatValidator.class),
+                mock(IntegerValidator.class),
+                mock(UrlValidator.class),
+                mock(RegexValidator.class),
+                mockTimeValidator,
+                Collections.emptySet()
         );
 
         Collection<Notice> result = underTest.validate(new ParsedEntity(
                 TEST_ID,
-                Map.of("unspecified_type_url", "http://mobilitydata.org",
-                        "string_type_url", "http://mobilitydata.org"),
+                Map.of("time_with_regex", "001:2:00003"),
                 new RawFileInfo.RawFileInfoBuilder().filename(TEST_FILE_TST).build()
         ));
 
-        assertEquals(0, result.size());
-
-        result = underTest.validate(new ParsedEntity(
-                TEST_ID,
-                Map.of("unspecified_type_url", "https://mobilitydata.org",
-                        "string_type_url", "https://mobilitydata.org"),
-                new RawFileInfo.RawFileInfoBuilder().filename(TEST_FILE_TST).build()
-        ));
+        assertEquals(1, result.size());
+        Notice notice = new ArrayList<>(result).get(0);
+        assertThat(notice, instanceOf(InvalidTimeNotice.class));
+        assertEquals("E016", notice.getId());
+        assertEquals("Invalid time", notice.getTitle());
+        assertEquals(TEST_FILE_TST, notice.getFilename());
+        assertEquals("Invalid time:001:2:00003 in field:time_with_regex for entity with id:test_id",
```

Comment:
See above - I'd suggest adding more times that should fail:
* "05:5:35"
* "05:05:5"
* "05:60:00"
* "05:00:60"

Reply:
Ok, in those unit tests the isValid return value is mocked and the parameter check is just here to verify it is passed unchanged. That way we don't test the validator library code at the same time. Do you think we shall have integration tests as well?